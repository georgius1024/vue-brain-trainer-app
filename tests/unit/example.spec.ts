import { shallowMount } from '@vue/test-utils';
import MenuItem from '@/components/menu-item.vue';
import router from '@/router';

describe('menu-item.vue', () => {
  it('renders correct way', () => {
    const to = 'test';
    const exact = false;
    const linkText = 'About';
    const propsData = { to, exact };

    const mountOptions = {
      propsData,
      router,
      slots: {
        default: linkText
      }
    };
    const wrapper = shallowMount(MenuItem, mountOptions);
    expect(wrapper.text()).toMatch(linkText);
    expect((wrapper.vm as any).className).toBeUndefined();
    router.push({ name: to });
    expect((wrapper.vm as any).className).toBe('uk-active');
    expect(wrapper).toMatchSnapshot();
  });
});
