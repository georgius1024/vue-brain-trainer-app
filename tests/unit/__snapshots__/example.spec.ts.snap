// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu-item.vue renders correct way 1`] = `
<li class="uk-active">
  <router-link-stub to="[object Object]" tag="a" event="click">About</router-link-stub>
</li>
`;
